project.mml

```diff
@@ -1226,7 +1226,9 @@ Layer:
             boundary,
             way_area/NULLIF(!pixel_width!::real*!pixel_height!::real,0) AS way_pixels
           FROM planet_osm_polygon
-          WHERE (boundary = 'national_park' OR leisure = 'nature_reserve')
+          WHERE (boundary = 'national_park'
+                 OR leisure = 'nature_reserve'
+                 OR (boundary = 'protected_area' AND tags->'protect_class' IN ('1','1a','1b','2','3','4','5','6','7','97','98','99')))
```

Comment:
I think it is not desirable to render `boundary=protected_area` + `protect_class=2`. As I understand https://wiki.openstreetmap.org/wiki/Tag:boundary%3Dprotected_area#Nature-protected-area such objects should be always tagged rather as `boundary=national_park`.

Classes `1`, `1a`, `1b`, `3`, `4` look like cases that should be tagged as  `leisure = nature_reserve` with potential adding also `boundary = protected_area` so rendering border based solely on `boundary = protected_area` + `protect_class=*` would encourage undesirable rendering (but I am less sure here).

Reply:
I am not familiar with this tagging scheme so let me know if I am misunderstanding something.

Reply:
I think nobody is familiar with all 3 tagging schemes, which would be a key to make any subtle connections. I don't remember too much after a year or so, even if I did a lot of research to compare, but my last conclusion was that it's a mess that can't be easily resolved:

https://github.com/gravitystorm/openstreetmap-carto/issues/603#issuecomment-350587768

Reply:
@matkoniecz: I understand text in wiki, that `protected_area` should tag **all** protected areas including national parks. Look at first sentence of this page:

> boundary=protected_area is used to mark boundaries of protected areas **including national parks**, marine protection areas, heritage sites, wilderness, indigenous and other, similar areas.

Reply:
@matkoniecz What do you think about this problem now?

Reply:
One object may not be tagged at once with `boundary=protected_area` and `boundary=national_park`. By rendering `boundary=protected_area` that should be tagged as `boundary=national_park` we basically allow `boundary=national_park` to be deprecated.

Reply:
Oh, I see now. I believe it's not deprecating, it's just description for one of valid (documented and used, I believe) tagging schemes. For me deprecation would be if there will be any reference which scheme should be used and which should not.

Still I feel this is mess that might be resolved at the tagging/documentation level, but now the best we could do now is to render all the "natural" classes. If anything will be more clear in the future, we can update the rendering.

Reply:
I will think a bit about this. For now I opened https://josm.openstreetmap.de/ticket/17111#ticket "convert boundary=protected_area + protect_class=2 into human readable equivalent boundary=national_park" (if someone thinks that boundary=national_park is not strictly preferable it would be a good idea to comment there).